feat: add filters for beam list views

Alchez opened the pull request:
Closes #210.

Commit:
fix: update deps

Commit:
fix: add base filters for manufacture

Commit:
fix: add base filters for demand

Commit:
fix: add base filters for receive

Commit:
fix: add base filters for ship

Commit:
fix: use inclusive filters

agritheory commented:
![image](https://github.com/user-attachments/assets/73f58b28-8754-4bd4-8097-ef09ae2dade1)

An excellent start. I think we can reduce the white space and switch to a two column layout. Another default filter that we had no included in the spec issue is "assigned" / "assigned to me".

Commit:
feat: add assigned filters

Alchez commented:
@agritheory A few updates:

- I've added a user filter for the `Manufacture`, `Demand`, `Receive` and `Ship` views.
  - It was straightforward for the Manufacture list view since that would use Frappe's resource API and the assignment data structure is documented. But I wasn't sure how assignments are stored in the demand DB.
  - I see an `assigned` string field inside the demand tables, but I'm not sure if that's supposed to represent a single user or a list of users (comma-separated?)
- I've also added a PR for making the filters a two-column layout over at https://github.com/agritheory/stonecrop/pull/243.
- I've added some design questions for scanning filters in the [connected issue](https://github.com/agritheory/beam/issues/210#issuecomment-2674255848)

agritheory commented:
z-index issue: 

![image](https://github.com/user-attachments/assets/330571e9-91af-4965-bef9-6f8b8ed3727d)

agritheory commented:
@Alchez 1) I think we need to z-index the container, this dropdown shouldn't be pushing the list down.
2) I think the break logic needs to be sync'd with the Stonecrop PR, where it's at 479px.

@crabinak FYI

![image](https://github.com/user-attachments/assets/45d5334d-4ab9-4f04-ac2a-59548ebd6a63)

Commit:
fix: update stonecrop dependencies

Commit:
Merge remote-tracking branch 'upstream/mobile_v15' into mobile_v15_filters

Commit:
fix: update dependencies

Alchez commented:
@agritheory @crabinak I added a fix for the dropdown pushing the list content over at https://github.com/agritheory/stonecrop/pull/250.

agritheory committed:
wip: scan output component

agritheory committed:
feat: make scan output configurable

Commit:
fix: minor fixes

agritheory commented on beam/beam/scan/config.py:
```diff
@@ -34,10 +34,8 @@ def get_scan_doctypes():
-		"BEAM Settings",
-		["show_scan_output"],
-	)
+	# TODO: should this be filtered against a specific company?
```
Associating an arbitrary user to company isn't easy, it requires an Employee record. I think adding a comment on the field is better that it will be visible in all companies.

agritheory commented:
@Alchez Not sure what happened in your cleanup, but the scan input component is now broken. These should be showing doctype and docname.

![image](https://github.com/user-attachments/assets/8e5e395f-dd3b-46b7-a96d-b63849f66abf)

Alchez commented:
@agritheory Hmm, I just looked at the code again and it seems like you're only returning those values (doctype: docname) for list action responses, whereas the `Manufacture` page is setup as a form page, which is why it doesn't get the `parenttype` and `parent` values from the server correctly.

agritheory commented:
@Alchez After updating the stonecrop packages, I am still seeing three filters on the same row but the z-index issue is fixed

![image](https://github.com/user-attachments/assets/b21cb988-1854-47e0-aec6-5fb6e658eb3d)

Alchez commented:
@agritheory Oh sorry, I didn't know we wanted to restrict the number of filters in each row as well. Should there be two filters per row maximum?
